jsonschema-validation.xml

```diff
@@ -179,15 +179,26 @@
 
             <section title="Validation of primitive types and child values">
                 <t>
-                    Two of the primitive types, array and object, allow for child values.  The validation of
-                    the primitive type is considered separately from the validation of child instances.
+                    Two of the primitive types, array and object, allow for child values.
+                    The validation of the primitive type is considered separately from
+                    the validation of child instances.
                 </t>
                 <t>
-                    For arrays, primitive type validation consists of validating restrictions on length.
+                    For arrays, primitive type validation consists of validating
+                    restrictions on length with "minItems" and "maxItems", while
+                    "items" and "additionalItems" determine which subschemas apply
+                    to which elements of the array.  In addition, "uniqueItems"
+                    and "contains" validate array contents as a whole.
                 </t>
                 <t>
-                    For objects, primitive type validation consists of validating restrictions on the presence
-                    or absence of property names.
+                    For objects, primitive type validation consists of validating
+                    restrictions on which and how many properties appear with
+                    "required", "minProperties", "maxProperties", "propertyNames",
+                    and the string array form of "dependencies", while "properties",
+                    "patternProperties", and "additionalProperties" determine which
+                    subschemas apply to which object property values.
```

Comment:
I'd consider adding "... and do not directly validate the instance itself.", after reading your explanation at https://github.com/json-schema-org/json-schema-spec/issues/161#issue-191384775 I found your phrasing there explaining...

> I recently rewrote the items, additionalItems, properties, patternProperties, and additionalProperties to make it clear that they do not directly validate the instance, but instead control which subschemas are applied to which child instances.

to actually be a clearer explanation that the changes you've made.